imageregistry/v1/types.go

```diff
@@ -268,6 +268,32 @@ type ImageRegistryConfigStorageAzure struct {
 	CloudName string `json:"cloudName,omitempty"`
 }
 
+// ImageRegistryConfigStorageIBMCOS holds the information to configure
+// the registry to use IBM Cloud Object Storage for backend storage.
+type ImageRegistryConfigStorageIBMCOS struct {
+	// bucket is the bucket name in which you want to store the registry's
+	// data.
+	// Optional, will be generated if not provided.
+	// +optional
+	Bucket string `json:"bucket,omitempty"`
+	// location is the IBM Cloud location in which your bucket exists.
+	// Optional, will be set based on the installed IBM Cloud location.
+	// +optional
+	Location string `json:"location,omitempty"`
+	// resourceGroupName is the name of the IBM Cloud resource group that this
+	// bucket is associated with.
+	// +optional
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
+	// resourceKeyCrn is the CRN of the IBM Cloud resource key that is created
+	// for the service instance. Required to contain HMAC type credentials.
+	// +optional
+	ResourceKeyCRN string `json:"resourceKeyCrn,omitempty"`
```

Comment:
What is this? I don't understand from the description.

Reply:
This is the CRN of an "resource key", aka service credentials for COS. I contemplated if I should name it "ServiceCredentialsCRN"..
The CRN looks like `crn:v1:bluemix:public:cloud-object-storage:global:a/xxxxx:yyyyy:resource-key:zzzzz`, so left it as "ResourceKeyCRN".

Reply:
You could add that this is effectively the "service credentials" CRN to the field description to help clarify.